src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -362,6 +364,27 @@ class ConcatExpression : public detail::VariadicExpression {
   }
 };
 
+// ENCODE_FOR_URI
+[[maybe_unused]] auto encodeForUriImpl =
+    [](std::optional<std::string> input) -> IdOrString {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  } else {
+    std::string value = input.value();
```

Comment:
No need to copy the full string.
I think you can use `std::string_view`, then also the `substr` below is efficient.
(I hope that `boost::url` supports it, otherwise you can still use a reference to the string).

Reply:
`boost::url` supports it and I changed two occurrences of `std::string` to `std::string_view` in the `encodeForUriImpl` function.